frontend/src/scenes/insights/views/InsightsTable/InsightsTable.tsx

```diff
@@ -278,17 +282,23 @@ export function InsightsTable({
                 CALC_COLUMN_LABELS.total
             ),
             render: function RenderCalc(_: any, item: IndexedTrendResult) {
-                let value
+                let value: number | undefined = undefined
                 if (calcColumnState === 'total' || isDisplayModeNonTimeSeries) {
                     value = item.count || item.aggregated_value
                 } else if (calcColumnState === 'average') {
                     value = average(item.data)
                 } else if (calcColumnState === 'median') {
                     value = median(item.data)
                 }
-                return item.action?.math_property
-                    ? formatPropertyValueForDisplay(item.action?.math_property, value)
-                    : (value ?? 'Unknown').toLocaleString()
+
+                return value !== undefined
+                    ? formatAggregationValue(
+                          item.action?.math_property,
+                          value,
+                          (value) => formatAggregationAxisValue(filters.aggregation_axis_format || 'numeric', value),
+                          formatPropertyValueForDisplay
+                      )
+                    : 'Unknown'
```

Comment:
I find this and the underlying function quite tough to follow... Not really a helpful comment as I don't have a great suggestion but I thought it worth saying anyways 🙈

Reply:
Yeah, that's fair... it's a yucky "do this or do that with some possibly specified default to fall back on"

The same (or at least very similar) idea is in multiple places in Insights so at the minute I've just been pushing it into one place.

Feels like maybe the problem is we pass around sets of numbers and then apply behaviour to them and we're seeing the start of there being a need to pass around sets of "things that have behaviour" 🤔